File: 17-More_Exceptions/more_exceptions.py
#Write your code here
class Calculator():
    def power(self, n, p):
        try:
            if n < 0 or p < 0:
                raise ValueError
            else:
                return n ** p
        except ValueError:
            return "n and p should be non-negative"

myCalculator=Calculator()
T=int(input())
for i in range(T):
    n,p = map(int, input().split())
    try:
        ans=myCalculator.power(n,p)
        print(ans)
    except Exception as e:
        print(e)

File: 17-More_Exceptions/more_exceptions.go
package main

import (
	"fmt"
	"math"
)

type Calculator struct{}

func (c Calculator) power(n, p float64) (float64, error) {
	if n < 0 || p < 0 {
		return 0, fmt.Errorf("n and p should be non-negative")
	}
	return math.Pow(n, p), nil
}

func main() {
	var inputs int
	_, err := fmt.Scan(&inputs)
	if err != nil {
		panic(err)
	}

	C := new(Calculator)
	for range inputs {
		var n, p float64
		_, err := fmt.Scan(&n, &p)
		if err != nil {
			panic(err)
		}

		if answer, err := C.power(n, p); err != nil {
			fmt.Println(err)
		} else {
			fmt.Println(answer)
		}
	}
}
